modules/realm_core/src/worker_thread_base.cpp

```diff
@@ -101,6 +101,19 @@ void WorkerThreadBase::run()
                "Timing [Total]: %lu ms",
                getCurrentTimeMilliseconds() - t);
     }
+
```

Comment:
If you add statistics at this position. they will also be computed with idle loops of the derived class. For example if the stages had nothing to process (queue empty) they will return "false" to the worker thread. The worker thread recognizes it in the if statement above, but then will jump into the statistics anyway, distorting the statistics with a timing of just a few milliseconds. So I guess it might be better to put the whole code into the `if(process()) { ... }` condition.

Reply:
I had started with it inside the if statement, then moved it outside.  For some reason I thought I had found some modules that would return false from process after it had done some work.  Looking back at the code I'm not seeing it today, so I may have been reading something wrong.  I will move it.